zebra-network/src/meta_addr.rs

```diff
@@ -521,6 +521,24 @@ impl MetaAddr {
         is_node && self.address_is_valid_for_outbound()
     }
 
+    /// Should this peer considered unreachable?
+    ///
+    /// A peer is considered unreachable if the last connection attempt to it failed and the last
+    /// successful connection is more than 3 days ago.
+    pub fn is_probably_unreachable(&self) -> bool {
+        self.last_connection_state == PeerAddrState::Failed && self.was_not_recently_seen()
+    }
+
+    /// Was this peer last seen a long time ago?
+    ///
+    /// A long time in this case is more than 3 days ago or never seen before.
+    pub fn was_not_recently_seen(&self) -> bool {
+        match self.last_seen() {
+            Some(last_seen) => last_seen.saturating_elapsed() > constants::MAX_PEER_TIME_UNSEEN,
+            None => true,
+        }
+    }
```

Comment:
I like the way these checks are written - they make sure we try each peer at least once.

### Edge Case

What do we want to happen if gossiped `untrusted_last_seen` times are a long way behind our local clock?

With this code, we'll try each peer once, and if it fails, we'll ignore it forever. That seems ok for now. But let's document it.

We might also want to review this design as part of #1873.

### Context

Outdated peer times can happen in at least four different ways:
* the user accidentally sets their clock to the next month or year
* a few peers have their clocks set to the previous month or year
* a peer implementation doesn't provide last seen times (and sets the field to zero)
* malicious peers deliberately gossip lots of peers with very old last seen times

Reply:
Done in [01d63d1](https://github.com/ZcashFoundation/zebra/pull/3030/commits/01d63d109500baa7056c8200983c78de4749dd3e). At least I think. Let me know if there's anything else I should do.

Reply:
I updated ticket #1873 with this edge case, I think we can deal with any remaining issues there.